app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    result_cache = {}
+
+    def inner(*args) -> Callable:
+        if args in result_cache:
```

Comment:
You should check both `args` and `kwargs` because the function might accept the same `args` but different `kwargs`. Additionally, I recommend checking the function name since there could be two different functions that accept the same arguments.